flytekit/common/tasks/task.py

```diff
@@ -199,7 +210,14 @@ def fetch_latest(cls, project, domain, name):
         :rtype: SdkTask
         """
         named_task = _common_model.NamedEntityIdentifier(project, domain, name)
-        admin_task = _engine_loader.get_engine().fetch_latest_task(named_task)
+        client = _flyte_engine._FlyteClientManager(
```

Comment:
I think this is increasing the blast radius of creating the client manager. I see this code fragment repeated again again and again.

I think 
`fetch` `register` etc should take in a context that has the client in it. But I understand that is a large change. In the interim have a helper method that creates the client, instead of passing the url and insecure flag repeatedly?

Reply:
https://github.com/lyft/flytekit/pull/153/commits/4a3604fb04acded509c2e157e634f85eabeeb5b4